Source/src/core/GameObject.cpp

```diff
@@ -310,6 +310,10 @@ void Hachiko::GameObject::Update()
 
     for (int i = 0; i < components.size(); ++i)
     {
+        if (!components[i]->IsActive())
```

Comment:
we should not have to do this here, at the beginning the Update method we should check that the current GO is active

Reply:
This has not been addressed yet, if we check that the GO is active, then we can skip this validation per component here